src/osdc/Objecter.h

```diff
@@ -1677,17 +1677,23 @@ class Objecter : public md_config_obs_t, public Dispatcher {
   std::map<int64_t, std::vector<pg_mapping_t>> pg_mappings;
 
   // convenient accessors
-  bool lookup_pg_mapping(const pg_t& pg, pg_mapping_t* pg_mapping) {
+  bool lookup_pg_mapping(const pg_t& pg, epoch_t epoch, std::vector<int> *up,
+                         int *up_primary, std::vector<int> *acting,
+                         int *acting_primary) {
     std::shared_lock l{pg_mapping_lock};
     auto it = pg_mappings.find(pg.pool());
     if (it == pg_mappings.end())
       return false;
     auto& mapping_array = it->second;
     if (pg.ps() >= mapping_array.size())
       return false;
-    if (mapping_array[pg.ps()].epoch != pg_mapping->epoch) // stale
+    if (mapping_array[pg.ps()].epoch != epoch) // stale
       return false;
-    *pg_mapping = mapping_array[pg.ps()];
+    auto& pg_mapping = mapping_array[pg.ps()];
+    up = pg_mapping.up;
```

Comment:
Missing dereference here and below -- make sure the thing compiles before pushing ;)

Reply:
Sorry. I usually compile and run tests on my machine before pushing.
It's just that currently I'm testing multiple PRs, so letting Jenkins here test it for me allows me to move faster, instead of serializing all of this work on my machine.
But anyway, I'll make sure to test compilation before I push :)